docker: fix pr-tests

bougue-pe opened the pull request:
A lot of things moved in the stack and were out of sync.
Also "fixed" compose "single-worker" mode.

🔍 review by (small) commit with comment is strongly advised

bougue-pe committed:
docker: use service name in all urls

Aim for "cattle" management, not "pet" management

Signed-off-by: Pierre-Etienne Bougué <bougue.pe@proton.me>

bougue-pe committed:
docker: just reorder docker compose to minimize diff vs. original compose

Signed-off-by: Pierre-Etienne Bougué <bougue.pe@proton.me>

bougue-pe committed:
docker: osrd-images uses 8081 in pr-tests, so let it free (in host mode)

Before this fix, openfga "squatted" it

Signed-off-by: Pierre-Etienne Bougué <bougue.pe@proton.me>

bougue-pe committed:
docker: fix (migrate) jaeger conf in pr-tests

broken after https://github.com/OpenRailAssociation/osrd/pull/11566

Signed-off-by: Pierre-Etienne Bougué <bougue.pe@proton.me>

bougue-pe committed:
docker: add dependencies in pr-tests

Signed-off-by: Pierre-Etienne Bougué <bougue.pe@proton.me>

bougue-pe committed:
docker: add minor conf in pr-tests to match original stack

Signed-off-by: Pierre-Etienne Bougué <bougue.pe@proton.me>

bougue-pe committed:
docker: fix editoast port in pr-tests for core after it was moved to 8089

Signed-off-by: Pierre-Etienne Bougué <bougue.pe@proton.me>

bougue-pe committed:
docker: fix "single-worker" mode

quick-fix forcing "host" mode as well

Signed-off-by: Pierre-Etienne Bougué <bougue.pe@proton.me>

bougue-pe committed:
docker: fix import of backup in pr-tests

Signed-off-by: Pierre-Etienne Bougué <bougue.pe@proton.me>

woshilapin commented on docker-compose.yml:
```diff
@@ -59,6 +59,7 @@ services:
       OPENFGA_DATASTORE_ENGINE: "postgres"
       OPENFGA_DATASTORE_URI: "postgres://osrd:password@osrd-postgres:5432/osrd?search_path=openfga"
       OPENFGA_HTTP_ADDR: "0.0.0.0:8091"
+      OPENFGA_GRPC_ADDR: "0.0.0.0:8093" # avoid squatting 8081
```
I would explicitely mention `osrd-images` in the comment, because if later, I end up on this comment, it might not be obvious why and who is using `8081`

bougue-pe replied:
Done in https://github.com/OpenRailAssociation/osrd/pull/11730/commits/22b4f1e443992c349af67378892f320fd25c061d

woshilapin commented on docker/docker-compose.pr-tests.yml:
```diff
@@ -188,7 +188,7 @@ services:
     container_name: osrd-jaeger-pr-tests
     restart: unless-stopped
-    command: --collector.otlp.grpc.host-port :4319 --collector.otlp.http.host-port :4320 --query.http-server.host-port :16687 --collector.grpc-server.host-port :14251
+    command: --set=receivers.otlp.protocols.grpc.endpoint=0.0.0.0:4319 --set=receivers.otlp.protocols.http.endpoint=0.0.0.0:4320 --set=receivers.jaeger.protocols.grpc.endpoint=0.0.0.0:14251 --set=extensions.jaeger_query.http.endpoint=0.0.0.0:16687
```
Will this work in non-host mode? (I'm talking about `0.0.0.0`.

bougue-pe replied:
Test on the actual pr-tests script is OK, and using the default address, so it should work as well as the "regular" stack.

leovalais commented on scripts/pr-tests-compose.sh:
```diff
@@ -56,6 +56,8 @@ if [ "$2" = "up" ] || [ "$2" = "up-and-load-backup" ]; then
         # Load backup
         export PR_TEST=1
         ./scripts/load-backup.sh "$3"
+        docker exec -it osrd-postgres-pr-tests psql -p 5433 -d osrd -c "create schema openfga;" || true
+        docker exec -it osrd-postgres-pr-tests psql -p 5433 -d osrd -c "grant all privileges on schema openfga to osrd;" || true
```
Maybe we can put that directly in `load-backup.sh`?

bougue-pe replied:
Yes, I checked locally that it is indeed useful in some case also on a regular compose.
Done in https://github.com/OpenRailAssociation/osrd/pull/11730/commits/d5618b0a5599abf0672b5bda878976899e5ab14d

leovalais reviewed (approved):
I'm no yaml engineering expert by any means and I never used pr-tests so here's the best review I can make: OpenFGA LGTM we don't use the GRPC interface 😁

flomonster commented on docker-compose.yml:
```diff
@@ -19,7 +19,7 @@ services:
       - "./docker/init_db.sql:/docker-entrypoint-initdb.d/init.sql"
     healthcheck:
-      test: [ "CMD", "pg_isready", "-d", "postgres://osrd:password@postgres/osrd" ]
+      test: [ "CMD", "pg_isready", "-d", "postgres://osrd:password@osrd-postgres/osrd" ]
```
Is there a reason to use the container name instead of the service name?

bougue-pe replied:
I'm not 100% sure, but I tried to make it consistent (osrdyne already uses jaeger's container-name, same for editoast with rabbit and osrdyne, same for core with editoast).
The idea behind is to avoid confusion/conflicts when multiple compose files are used (which would silently break if on the same network?).

Maybe we should just use different service names for pr-tests compose file ? Not sure what's best/useful here.

Khoyo replied:
> which would silently break if on the same network?

Multiple compose files on the same (docker) network is asking for trouble, and won't happen unless we explicitly ask for it (eg. an external network definition). AFAIK docker won't do any dns resolution across networks. 

Since we force container names anyway, I don't see any meaningful difference between the two options. If we didn't, I'd have a preference for the service name.

bougue-pe replied:
Switched back to full service-names in https://github.com/OpenRailAssociation/osrd/pull/11730/commits/a8b28185aee1bd37b7ff5b5b8b6803d3d99b27bd (amended and squashed since)

Testing multiple stacks running together lead to some discoveries and fixes: please review additional commits (by commit starting from https://github.com/OpenRailAssociation/osrd/pull/11730/commits/14d5a02d16ba170631d5764f6c0a2f83488e364d is easier).
One issue is open and will stay open after this PR: https://github.com/OpenRailAssociation/osrd/issues/11871

bougue-pe committed:
docker: pr-tests: improve network and worker prefix

Signed-off-by: Pierre-Etienne Bougué <bougue.pe@proton.me>

bougue-pe committed:
docker: cleanup-db: fix container search to match exact name

Ex: if pr-tests stack is running, 'osrd-editoast-pr-tests' shouldn't be
matched by a search of 'osrd-editoast'.

Signed-off-by: Pierre-Etienne Bougué <bougue.pe@proton.me>

bougue-pe committed:
docker: osrdyne: specify exact worker image name for docker-driver

Signed-off-by: Pierre-Etienne Bougué <bougue.pe@proton.me>

bougue-pe committed:
docker: pr-tests: document a bit use-cases (test-cases)

NB: At the time of this commit, pr-tests stack is broken when working
along a regular stack (osrdyne concurrency).

Signed-off-by: Pierre-Etienne Bougué <bougue.pe@proton.me>

Khoyo commented on docker/docker-compose.pr-tests.yml:
```diff
@@ -65,12 +67,16 @@ services:
+      OPENFGA_GRPC_ADDR: "0.0.0.0:8193" # avoid squatting 8081 (used by osrd-images in pr-tests)
       OPENFGA_PLAYGROUND_PORT: 8192
     command: run
     ports:
```
The published ports should probably be different  in pr-tests

bougue-pe replied:
It looks OK, no?
8`0`91/8`0`92 for regular stack
8`1`91/8`1`92 for pr-tests

The 8093 and 8193 are just declared to avoid squatting 8081 especially in host mode (so more important for regular stack but consistent between stacks).

Khoyo replied:
You're right, I didn't notice the `0`/`1` :D

Khoyo reviewed (approved):
Otherwise LGTM
